### ISSUE TITLE
Check more carefully what library is used for the analysis

### ISSUE THREAD

**Issue by piever** [piever/Sputnik.jl#12]
Choosing between StatPlots, GroupedErrors and OnlineStats

**Comment by shashi** [comment]
This is where I feel it would help to think about API similar to RecipesBase.jl would be useful. These libs should be able to depend on a pure Julia SputnikBase.jl, use a macro to describe the types of analyses and the options they take, Sputnik can then take that specification and implement how they get displayed...

It's an ambiguous goal, so I'm ok with doing whatever is easy for now.

Cc @joshday

**Comment by piever** [comment]
I fully agree with the ambitious plan. I've so far (very preliminarily) defined a `Analysis` struct [here](https://github.com/piever/Sputnik.jl/blob/master/src/process.jl#L1) and I'm doing the same as Plots type recipes, meaning that I overload `process(::GroupedError, ::Analysis)` and `process(::StatPlotsRecipe, ::Analysis)`. Eventually this should be moved in a low dependency package that GroupedErrors, StatPlots and OnlineStats are happy to depend on.

The interface is clearly very preliminary and I'd love yours and Josh's input on it.
